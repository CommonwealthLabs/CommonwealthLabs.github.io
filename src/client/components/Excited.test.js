// import test from 'ava';

import React from 'react';
// import register from 'ignore-styles'
// register(['.sass', '.scss'])
import { ExcitedComponent as Excited, mapStateToProps } from './Excited.react';
import { MemoryRouter } from 'react-router';
import sinon from 'sinon';
// import sum from './sum';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('<Excited />: ', () => {
  let callback = sinon.spy();

  const wrapper = shallow(<Excited dispatch={callback} exclamation="!!!" />);

  it('should render as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should trigger a callback when <button/> is clicked', () => {
    // manually trigger the callback
    wrapper.find('button').simulate('click');
    expect(callback.called).toBe(true);
  });

  it('should retrieve appropriate props from state', () => {
    let testState = {
      exclamation: '!!'
    };

    let expectedState = {
      exclamation: '!!'
    };

    expect(JSON.stringify(mapStateToProps(testState))).toBe(
      JSON.stringify(expectedState)
    );
  });
});
