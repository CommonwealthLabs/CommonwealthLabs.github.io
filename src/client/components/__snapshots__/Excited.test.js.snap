// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Excited />:  should render as expected 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h1>
      LOL!!!
    </h1>
    <button
      onClick={[Function]}
    >
      Get Excited!
    </button>
    <Link
      replace={false}
      to="/whatever"
    >
      Go Here
    </Link>
  </div>,
  "nodes": Array [
    <div>
      <h1>
        LOL!!!
      </h1>
      <button
        onClick={[Function]}
      >
        Get Excited!
      </button>
      <Link
        replace={false}
        to="/whatever"
      >
        Go Here
      </Link>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Unknown
      dispatch={[Function]}
      exclamation="!!!"
    />,
    "_instance": null,
    "_newState": null,
    "_rendered": <div>
      <h1>
        LOL!!!
      </h1>
      <button
        onClick={[Function]}
      >
        Get Excited!
      </button>
      <Link
        replace={false}
        to="/whatever"
      >
        Go Here
      </Link>
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown
    dispatch={[Function]}
    exclamation="!!!"
  />,
}
`;
